js/piwik.js

```diff
@@ -5264,12 +5264,6 @@ if (typeof window.Piwik !== 'object') {
                 // and visitorId (eg userId might be set etc)
                 link = removeUrlParameter(link, configVisitorIdUrlParameter);
 
-                if (link.indexOf('?') > 0) {
```

Comment:
would you still need to do this logic when there is no hash in the URL?

Reply:
No - the logic within the **addUrlParameter** function duplicates the removed logic, but applies to the **baseURL** as opposed to the full **url**:
```
            if (baseUrl.indexOf('?') === -1) {
                baseUrl += '?';
            } else if (!stringEndsWith(baseUrl, '?')) {
                baseUrl += '&';
            }
            // nothing to if ends with ?
```
The **baseURL** variable will be the same as the **url** variable in the case that there are no hashes in the url.

As the **addUrlParameter** function is always called from the **replaceHrefForCrossDomainLink** directly after the removed logic, then the inclusion of this logic is superfluous in all cases. 

I have tested the 4 use cases identified under **Example Destination Links** in the description above, to ensure that they are processed correctly to generate the **Expected Results**

Reply:
makes sense 👍